src/main/java/org/rumbledb/cli/JsoniqQueryExecutor.java

```diff
@@ -94,15 +95,15 @@ public void runQuery(String queryFile, String outputPath) throws IOException {
 
         if (result.isRDD() && outputPath != null) {
             JavaRDD<Item> rdd = result.getRDD(dynamicContext);
-            JavaRDD<String> output = rdd.map(o -> o.serialize());
-            output.saveAsTextFile(outputPath);
+            JavaRDD<String> outputRDD = rdd.map(o -> o.serialize());
+            outputRDD.saveAsTextFile(outputPath);
         } else {
-            String output = getIteratorOutput(result, dynamicContext);
+            outputList = getIteratorOutput(result, dynamicContext);
```

Comment:
If I understand this correctly, getIteratorOutput here will truncate the local results prior to writing the results to the file. I think the issue #596 revolves around this point.

Reply:
Yes, very nice that you opened the issue. It would indeed make sense to not limit materialization if this is the final output. This should be handled separately of the server.